File: food-ordering-system/src/app/shared/models/Price.ts
import {Product} from "./Product";

export type PricePure = {
  id: string;
  value: number;
  previousValue: number;
  productId: string;
  comment: string;
}

export type Price = PricePure & {
  product?: Product;
}


export type PriceRequest = {
  id?: string;
  value: number;
  previousValue: number;
  productId: string;
  comment: string;
}

File: FoodOrderingApp/Public.DTO/v1/Price.cs
using Base.Contracts.Domain;

namespace Public.DTO.v1;

public class PricePure : IDomainEntityId
{
    public Guid Id { get; set; }
    public decimal Value { get; set; }
    public decimal? PreviousValue { get; set; }
    public Guid ProductId { get; set; }
    public string? Comment { get; set; }
}

public class Price : PricePure
{
    public Product? Product { get; set; }
}

public class PriceRequest
{
    public Guid? Id { get; set; }
    public decimal Value { get; set; }
    public decimal? PreviousValue { get; set; }
    public Guid ProductId { get; set; }
    public string? Comment { get; set; }
}